sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionContainer.java

```diff
@@ -0,0 +1,199 @@
+                                                CosmosItemRequestOptions requestOptions) {
+        return blockItemResponse(this.cosmosEncryptionAsyncContainer.upsertItem(item, partitionKey, requestOptions));
+    }
+
+    /**
+     * replaces item and encrypts the requested fields
+     *
+     * @param item           the Cosmos item represented as a POJO or Cosmos item object.
+     * @param itemId         the item id.
+     * @param partitionKey   the partition key.
+     * @param requestOptions request option
+     * @param <T>            serialization class type
+     * @return result
+     */
+    @SuppressWarnings("unchecked")
+    public <T> CosmosItemResponse<T> replaceItem(T item,
+                                                 String itemId,
+                                                 PartitionKey partitionKey,
+                                                 CosmosItemRequestOptions requestOptions) {
+        return blockItemResponse(this.cosmosEncryptionAsyncContainer.replaceItem(item, itemId, partitionKey,
+            requestOptions));
+    }
+
+    /**
+     * Reads item and decrypt the encrypted fields
+     *
+     * @param id             item id
+     * @param partitionKey   the partition key.
+     * @param requestOptions request options
+     * @param classType      deserialization class type
+     * @param <T>            type
+     * @return result
+     */
+    public <T> CosmosItemResponse<T> readItem(String id,
+                                              PartitionKey partitionKey,
+                                              CosmosItemRequestOptions requestOptions,
+                                              Class<T> classType) {
+        return blockItemResponse(this.cosmosEncryptionAsyncContainer.readItem(id, partitionKey, requestOptions,
+            classType));
+    }
+
+    /**
+     * Query for items in the current container using a string.
+     *
+     * @param <T>       the type parameter.
+     * @param query     the query text.
+     * @param options   the query request options.
+     * @param classType the class type.
+     * @return a {@link CosmosPagedIterable}.
+     */
+    public <T> CosmosPagedIterable<T> queryItems(String query, CosmosQueryRequestOptions options,
+                                                 Class<T> classType) {
+
+        return getCosmosPagedIterable(this.cosmosEncryptionAsyncContainer.queryItems(query, options, classType));
+    }
+
+    /**
+     * Query for items in the current container using a {@link SqlQuerySpec}.
+     *
+     * @param <T>       the type parameter.
+     * @param query     the query.
+     * @param options   the query request options.
+     * @param classType the class type.
+     * @return a {@link CosmosPagedIterable}.
+     */
+    public <T> CosmosPagedIterable<T> queryItems(SqlQuerySpec query, CosmosQueryRequestOptions options,
+                                                 Class<T> classType) {
```

Comment:
nit: in the other method signatures you have one arg per line. can we follow that here as well?

Reply:
done